src/plugins/printer.ts

```diff
@@ -0,0 +1,76 @@
+import {Plugin, Cordova} from './plugin';
+declare var cordova: any;
+
+export interface PrintOptions {
+    /**
+     * The name of the print job and the document
+     */
+    name?: string;
+
+    /**
+     * The network URL of the printer. 
+     * Only supported on iOS.
+     */
+    printerId?: string;
+
+    /**
+     * Specifies the duplex mode to use for the print job.
+     * Either double-sided (duplex:true) or single-sided (duplex:false).
+     * Double-sided by default.
+     * Only supported on iOS
+     */
+    duplex?: boolean;
+
+    /**
+     * The orientation of the printed content, portrait or landscape
+     * Portrait by default.
+     */
+    landscape?: boolean;
+
+    /**
+     * If your application only prints black text, setting this property to true can result in better performance in many cases.
+     * False by default.
+     */
+    grayscale?: boolean;
+
+    /**
+     * The Size and position of the print view
+     */
+    bounds?: number[] | any;
+}
+
+
+@Plugin({
+    plugin: 'de.appplant.cordova.plugin.printer',
+    pluginRef: 'cordova.plugins.printer',
+    repo: 'https://github.com/katzer/cordova-plugin-printer.git',
+    platforms: ['Android', 'iOS']
+})
+export class Printer {
+
+    /**
+     * Checks whether to device is capable of printing.
+     */
+    @Cordova()
+    static isAvailable(): Promise<boolean> {
+        return new Promise((resolve, reject) => {
```

Comment:
No need to return a promise manually. Also the original plugin doesn't have Promise support.

The `@Cordova` decorator overrides your function anyways. So this will still work. But you should replace the return here and below with just `return; `. 

The original plugin usage is: 
`isAvailable(callback)`